### ISSUE TITLE
The task close by itself when selecting anything in a task

### ISSUE THREAD

**Issue by rubenkenobi** [kanboard/kanboard#5288]
when modifying, altering, renaming a task by selecting the typed line using the mouse, clicking and dragging, the task closes and returns to the frame without the modifications.

**Comment by mwlpeeters** [comment]
This is what I'm experiencing right now with the latest release:

When you click within the popup, hold the mouse button (to drag and select) and you release the button outside of the popup it will close the popup.
The click event outside the popup should only trigger if the button down event was originated outside the popup window.

**Comment by tomeli5n** [comment]
What version are you using ?

This was fixed on https://github.com/kanboard/kanboard/releases/tag/v1.2.29
>Do not close modals when clicking on the background if the form has changed

**Comment by mwlpeeters** [comment]
I have this with version v1.2.30.
What was fixed is closing the window when there are changes. Now it only closes when there are no modifications.
But it still closes pop-ups when you select the text using a dragging motion and release the mouse outside the pop-up.
For example open a task, select the title in the text box from right to left and release the mouse button just outside of the pop-up, it will close.

**Comment by mwlpeeters** [comment]
@fguillot do you still need more information on this issue?

**Comment by nekohayo** [comment]
Not just that, it happens in all cases when you are doing any click's release "outside of the modal" as part of a drag (which is the case when you are making text selections).

So, in the latest version, if you start selecting text with the mouse by click-and-drag, and you end your selection outside the window (ex: on the left), which is quite frequent when you are selecting a paragraph's worth of text and you want to select "whole lines" (especially if you are going from bottom/right to top/left as a general direction), then the modal closes _even if you had unsaved changes in the textfield_ ...

**Comment by nekohayo** [comment]
Indeed it is unexpected that the popup would close when a mouse release event occurs outside in the background, which is nearly inevitable when making text line selections because the textfield is right by the edge of the popup (you merely have what seems like a ~5px total margin for error, which is very difficult to hit accurately, even without accessibility problems… I'd imagine it would be much worse for someone who doesn't have fine motor skills).

Just for the record / to be clear about what I mean, here's a very short demonstration of the problematic behavior I frequently run into, which causes me to lose unsaved changes:

https://github.com/kanboard/kanboard/assets/479401/7ac01719-97fc-4c8b-a8ca-52efa3236f07

This happens both within the kanban columns view, and within the card's detailed view (as long as you click "Edit the task" to call the popup).

I agree that clicking (or releasing a mouse click/drag) "on the background" outside of the popup should not dismiss the popup, if that's the most straightforward fix; generally speaking, I think it would be much safer to make dismissing the popup the result of explicit actions only with the Save / Cancel / X buttons (or corresponding keyboard shortcuts, like the existing `Esc` shortcut) anyway. Thanks :)

**Comment by nekohayo** [comment]
I see this has been released as v1.2.33 now. I have tested that new version on my personal kanbans and it solves all the problems for me, no more accidents! Thank you 🙂
